hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/DiscoveryJoiner.java

```diff
@@ -52,6 +66,7 @@ public DiscoveryJoiner(Node node, DiscoveryService discoveryService, boolean use
         this.maximumWaitingTimeBeforeJoinSeconds = node.getProperties().getInteger(WAIT_SECONDS_BEFORE_JOIN);
         this.discoveryService = discoveryService;
         this.usePublicAddress = usePublicAddress;
+        this.discoveryPublicAddressFallback = Boolean.getBoolean(DISCOVERY_PUBLIC_ADDRESS_FALLBACK_PROPERTY);
```

Comment:
It seems that it's should be `node.getProperties().getBoolean(DISCOVERY_PUBLIC_ADDRESS_FALLBACK_PROPERTY)` here

Reply:
since that's an internal property to use in case of regression, we can just keep it as system property like we do for other similar cases e.g. https://github.com/hazelcast/hazelcast/blob/7c3894b1474dba56648cf10ad8441615486fcdea/hazelcast/src/main/java/com/hazelcast/internal/nio/ClassLoaderUtil.java#L48, wdyt?

Reply:
LGTM